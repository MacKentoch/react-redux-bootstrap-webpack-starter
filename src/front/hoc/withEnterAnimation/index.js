// @flow

import withEnterAnimation from './withEnterAnimation';

export default withEnterAnimation;
