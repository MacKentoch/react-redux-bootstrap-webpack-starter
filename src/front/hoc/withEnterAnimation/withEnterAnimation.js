// @flow

// #region imports
import React, { Component } from 'react';
import wrapDisplayName from 'recompose/wrapDisplayName';
import AnimatedDiv from './styled/AnimatedDiv';
// #endregion

// #region flow types
type Props = any;

type State = any;
// #endregion

function withEnterAnimation() {
  return (BaseComponent: any) => {
    class WithEnterAnimation extends Component<Props, State> {
      render() {
        const { ...passProps } = this.props;

        return (
          <AnimatedDiv viewEnter={true}>
            <BaseComponent {...passProps} />
          </AnimatedDiv>
        );
      }
    }

    /* eslint-disable no-process-env */
    if (process.env.NODE_ENV !== 'production') {
      // HOC would obfuscate component name, this trick is helpful for dev (we don't care in production)
      WithEnterAnimation.displayName = wrapDisplayName(
        BaseComponent,
        'withEnterAnimation',
      );
    }
    /* eslint-enable no-process-env */

    return WithEnterAnimation;
  };
}

export default withEnterAnimation;
