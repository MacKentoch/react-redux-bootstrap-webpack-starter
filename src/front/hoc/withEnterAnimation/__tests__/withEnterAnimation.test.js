// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import withEnterAnimation from '../withEnterAnimation';
// #endregion

describe('withEnterAnimation HOC', () => {
  it('renders as expected', () => {
    const DummyComponent = () => <p>component</p>;

    const component = renderer
      .create(withEnterAnimation()(DummyComponent))
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
