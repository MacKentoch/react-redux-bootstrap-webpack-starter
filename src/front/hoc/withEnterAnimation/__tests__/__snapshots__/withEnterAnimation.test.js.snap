// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withEnterAnimation HOC renders as expected 1`] = `null`;
